hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -521,7 +521,7 @@ object HoodieSparkSqlWriter {
                                              jsc: JavaSparkContext,
                                              tableInstantInfo: TableInstantInfo
                                              ): (Boolean, common.util.Option[java.lang.String], common.util.Option[java.lang.String]) = {
-    if(writeResult.getWriteStatuses.rdd.filter(ws => ws.hasErrors).isEmpty()) {
+    if(writeResult.getWriteStatuses.rdd.filter(ws => ws.hasErrors).count() == 0) {
```

Comment:
nit: `if (writeResult.xxx)`

Reply:
if the writeStatuses are cached/persisted, would that still trigger recomputations?